clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/DeliverArraysExecutor.scala

```diff
@@ -0,0 +1,62 @@
+package org.broadinstitute.clio.client.dispatch
+
+import org.broadinstitute.clio.client.commands.DeliverArrays
+import org.broadinstitute.clio.client.util.IoUtil
+import org.broadinstitute.clio.transfer.model.Metadata
+import org.broadinstitute.clio.transfer.model.arrays.{
+  ArraysExtensions,
+  ArraysMetadata
+}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * Special-purpose CLP for delivering arrays to FireCloud workspaces.
+  *
+  * Wraps the move CLP with extra IO / upsert logic:
+  *
+  *   1. Writes the arrays md5 value to file at the target path
+  *   2. Records the workspace name in the metadata for the delivered arrays
+  */
+class DeliverArraysExecutor(deliverCommand: DeliverArrays)
+    extends DeliverExecutor(deliverCommand) {
+
+  override def customMetadataOperations(
+    metadata: ArraysMetadata,
+    ioUtil: IoUtil
+  )(
+    implicit ec: ExecutionContext
+  ): Future[ArraysMetadata] = {
+    metadata.grnIdat match {
```

Comment:
Seems like "copy instead of move" is sort of a general pattern that could be abstracted if it's not a pain